src/EStimLibrary/Core/ReusableIdPool.cs

```diff
@@ -74,16 +86,24 @@ public SortedSet<int> FreeIds
     /// negative number is given. Can be changed after construction.</param>
     public ReusableIdPool(int baseId, int numIds)
     {
-        // Store the base ID. Property setter handles defaulting to 0.
-        this.BaseId = baseId;
+        // Ensure BaseId is not negative
+        this.BaseId = Math.Max(baseId, MIN_BASE_ID);
```

Comment:
this check is already done via the protected set() method of the BaseId property (see above, around line 41), so you can remove it here